WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -0,0 +1,355 @@
+        })
+    }
+
+    override fun onNavigationItemReselected(item: MenuItem) {
+        // scroll the active fragment's contents to the top when user re-taps the current item
+        val position = getPositionForItemId(item.itemId)
+        if (position != PAGE_NEW_POST) {
+            (navAdapter?.getFragment(position) as? OnScrollToTopListener)?.onScrollToTop()
+        }
+    }
+
+    private fun getPositionForItemId(@IdRes itemId: Int): Int {
+        return when (itemId) {
+            R.id.nav_sites -> PAGE_MY_SITE
+            R.id.nav_reader -> PAGE_READER
+            R.id.nav_write -> PAGE_NEW_POST
+            R.id.nav_me -> PAGE_ME
+            else -> PAGE_NOTIFS
+        }
+    }
+
+    @IdRes
+    private fun getItemIdForPosition(position: Int): Int {
+        return when (position) {
+            PAGE_MY_SITE -> R.id.nav_sites
+            PAGE_READER -> R.id.nav_reader
+            PAGE_NEW_POST -> R.id.nav_write
+            PAGE_ME -> R.id.nav_me
+            else -> R.id.nav_notifications
+        }
+    }
+
+    private fun setCurrentPosition(position: Int, ensureSelected: Boolean) {
+        // new post page can't be selected, only tapped
+        if (position == PAGE_NEW_POST) {
+            return
+        }
+
+        // remove the title and selected state from the previously selected item
+        if (prevPosition > -1) {
+            showTitleForPosition(prevPosition, false)
+            setImageViewSelected(prevPosition, false)
+        }
+
+        // set the title and selected state from the newly selected item
+        showTitleForPosition(position, true)
+        setImageViewSelected(position, true)
+
+        AppPrefs.setMainPageIndex(position)
+        prevPosition = position
+
+        if (ensureSelected) {
+            // temporarily disable the nav listeners so they don't fire when we change the selected page
+            assignNavigationListeners(false)
+            try {
+                selectedItemId = getItemIdForPosition(position)
+            } finally {
+                assignNavigationListeners(true)
+            }
+        }
+
+        val fragment = navAdapter.getFragment(position)
+        if (fragment != null) {
+            fragmentManager
+                    .beginTransaction()
+                    .replace(R.id.fragment_container, fragment)
+                    .setTransition(TRANSIT_FRAGMENT_FADE)
+                    .commit()
+        }
+    }
+
+    /*
+     * ideally we'd use a color selector to tint the icon based on its selected state, but prior to
+     * API 21 setting a color selector via XML will crash the app, and setting it programmatically
+     * will have no effect
+     */
+    private fun setImageViewSelected(position: Int, isSelected: Boolean) {
+        val color = resources.getColor(if (isSelected) R.color.blue_medium else R.color.grey_lighten_10)
+        getImageViewForPosition(position)?.setColorFilter(color, android.graphics.PorterDuff.Mode.MULTIPLY)
+    }
```

Comment:
You can get around the deprecated `resources.getColor(...)` by using `ContextCompat.getColor(...)`
```
getImageViewForPosition(position)?.setColorFilter(ContextCompat
                .getColor(context, (if (isSelected) R.color.blue_medium else R.color.grey_lighten_10)),
                android.graphics.PorterDuff.Mode.MULTIPLY)
```